google/gax/api_callable.py

```diff
@@ -41,59 +41,63 @@
 _MILLIS_PER_SECOND = 1000
 
 
-def _add_timeout_arg(a_func, timeout):
-    """Updates a_func so that it gets called with the timeout as its final arg.
+class _RetryableWrapper(object):
```

Comment:
I don't feel strongly about this, but I don't think this is particularly unnaturally expressed as a function either -- it seems to me that the only "object-y" thing about this is storing the read-only attribute `settings`, which works equally well in a closure.

I'm fine with it either way, although I would rather `settings` belonged to a closure (or callable object) in `create_api_call` than in the retrying part of the code.